llvm/lib/Analysis/TargetLibraryInfo.cpp

```diff
@@ -1149,6 +1149,18 @@ bool TargetLibraryInfoImpl::getLibFunc(const Function &FDecl,
   return isValidProtoForLibFunc(*FDecl.getFunctionType(), F, *M);
 }
 
+bool TargetLibraryInfoImpl::getLibFunc(const Instruction &I, LibFunc &F) const {
+  if (I.getOpcode() != Instruction::FRem)
+    return false;
+
+  Type *ScalarTy = I.getType()->getScalarType();
```

Comment:
I must have missed this before but this should just be `I.getType();` because otherwise you'll return a value for vector types, which would be wrong.